apimodels/agent_models.go

```diff
@@ -143,20 +143,21 @@ type CreateHost struct {
 	Retries             int    `mapstructure:"retries" json:"retries" yaml:"retries"`
 
 	// EC2-related settings
-	AMI             string      `mapstructure:"ami" json:"ami" yaml:"ami" plugin:"expand"`
-	Distro          string      `mapstructure:"distro" json:"distro" yaml:"distro" plugin:"expand"`
-	EBSDevices      []EbsDevice `mapstructure:"ebs_block_device" json:"ebs_block_device" yaml:"ebs_block_device" plugin:"expand"`
-	InstanceType    string      `mapstructure:"instance_type" json:"instance_type" yaml:"instance_type" plugin:"expand"`
-	IPv6            bool        `mapstructure:"ipv6" json:"ipv6" yaml:"ipv6"`
-	Region          string      `mapstructure:"region" json:"region" yaml:"region" plugin:"expand"`
-	SecurityGroups  []string    `mapstructure:"security_group_ids" json:"security_group_ids" yaml:"security_group_ids" plugin:"expand"`
-	Spot            bool        `mapstructure:"spot" json:"spot" yaml:"spot"`
-	Subnet          string      `mapstructure:"subnet_id" json:"subnet_id" yaml:"subnet_id" plugin:"expand"`
-	UserdataFile    string      `mapstructure:"userdata_file" json:"userdata_file" yaml:"userdata_file" plugin:"expand"`
-	UserdataCommand string      `json:"userdata_command" yaml:"userdata_command" plugin:"expand"`
-	AWSKeyID        string      `mapstructure:"aws_access_key_id" json:"aws_access_key_id" yaml:"aws_access_key_id" plugin:"expand"`
-	AWSSecret       string      `mapstructure:"aws_secret_access_key" json:"aws_secret_access_key" yaml:"aws_secret_access_key" plugin:"expand"`
-	KeyName         string      `mapstructure:"key_name" json:"key_name" yaml:"key_name" plugin:"expand"`
+	AMI                   string      `mapstructure:"ami" json:"ami" yaml:"ami" plugin:"expand"`
+	Distro                string      `mapstructure:"distro" json:"distro" yaml:"distro" plugin:"expand"`
+	EBSDevices            []EbsDevice `mapstructure:"ebs_block_device" json:"ebs_block_device" yaml:"ebs_block_device" plugin:"expand"`
+	InstanceType          string      `mapstructure:"instance_type" json:"instance_type" yaml:"instance_type" plugin:"expand"`
+	IPv6                  bool        `mapstructure:"ipv6" json:"ipv6" yaml:"ipv6"`
+	Region                string      `mapstructure:"region" json:"region" yaml:"region" plugin:"expand"`
+	IAMInstanceProfileARN string      `mapstructure:"iam_instance_profile_arn" json:"iam_instance_profile_arn" yaml:"iam_instance_profile_arn" plugin:"expand"`
```

Comment:
Is this meant to be used on `host.create` hosts as well? If so, we want to add this to the `host.create` docs, as well as to Shrub.

Reply:
On second though, I don't think we need it for this use case. removed.